### DISCUSSION TITLE
How can i read a text file in wing?

### DISCUSSION THREAD

**Discussion by schosterbarak** [winglang/wing#2547] [Q&A] [answered]
Hey have a text file i'd like to read in my winglang function.
The text is in the same directory as the `.w' file
what's the right way to reference and read it?

**Comment by staycoolcall911** [comment, marked as answer]
Hey, I think for now you should use `extern` to call external javascript, then use an fs package to read the file.
Just in case you're unfamiliar with `extern`, you can see some examples [here](https://github.com/winglang/wing/blob/main/examples/tests/valid/extern_implementation.w). Note that the external javascript should reside in a separate file (in the example I sent the external js is [here](https://github.com/winglang/wing/blob/main/examples/tests/valid/external_js.js)).
